File: DataStructures/Stack/Python/stack.py
from typing import TypeVar, Generic

T = TypeVar('T')

class stack(Generic[T]):

    def __init__(self) -> None:
        self.items = []

    def is_empty(self) -> bool:
        return len(self.items) == 0
    
    def push(self, item: T) -> None:
        self.items.append(item)
    
    def pop(self) -> T:
        return self.items.pop()
    
    def peek(self) -> T:
        if not self.is_empty:
            return self.items[-1]
        return None
    
    def size(self) -> int:
        return len(self.items)
    
    def to_string(self) -> str:
        return str(self.items)

File: DataStructures/Stack/Typescript/stack.ts
export class Stack<T> {
    items: T[];

    constructor() {
        this.items = [];
    }

    isEmpty(): boolean {
        return this.items.length == 0;
    }
    
    push(item: T): void {
        this.items.push(item);
    }

    pop(): T {
        let item = null;
        if (this.isEmpty) {
            return null;
        }
        return this.items.pop();
    }

    peek(): T | void {
        if (this.isEmpty()) {
            return null;
        }
        return this.items[this.items.length-1];
    }

    size(): number {
        return this.items.length;
    }

    toString(): string {
        return this.items.toString();
    }
}
